crun.1

```diff
@@ -707,9 +707,78 @@ These flags are supported:
 
 .SH idmap mount options
 .PP
-If the \fB\fCidmap\fR option is specified then the mount is ID mapped using the container
-target user namespace.  This is an experimental feature and can change at any time
-without notice.
+If the \fB\fCidmap\fR option is specified then the mount is ID mapped using
+the container target user namespace.  This is an experimental feature
+and can change at any time without notice.
+
+.PP
+The \fB\fCidmap\fR option supports a custom mapping that can be different
+than the user namespace used by the container.
+
+.PP
+The mapping can be specified after the \fB\fCidmap\fR option like:
+\fB\fCidmap=uids=0-1-10;gids=0-100-10\fR\&.
+
+.PP
+For each triplet, the first value is the start of the backing
+filesystem IDs that are mapped to the second value on the host.  The
+length of this mapping is given in the third value.
+
+.PP
+These values are written to the \fB\fC/proc/$PID/uid_map\fR and
+\fB\fC/proc/$PID/gid_map\fR files to create the user namespace for the
+idmapped mount.
+
+.PP
+The only two options that are currently supported are \fB\fCuids\fR and
+\fB\fCgids\fR\&.
+
+.PP
+When a custom mapping is specified, a new user namespace is created
+for the idmapped mount.
+
+.PP
+If no option is specified, then the container user namespace is used.
+
+.PP
+If the specified mapping is prepended with a '@' then the mapping is
+considered relative to the container user namespace.  The host ID for
+the mapping is changed to account for the relative position of the
+container user in the container user namespace.
+
+.PP
+For example, the mapping: \fB\fCuids=@1-3-10\fR, given a configuration like
+
+.PP
+.RS
+
+.nf
+"uidMappings": [
+      {
+        "containerID": 0,
+        "hostID": 0,
+        "size": 1
+      },
+      {
+        "containerID": 1,
+        "hostID": 2,
+        "size": 1000
+      }
+    ]
+
+.fi
+.RE
+
+.PP
+will be converted to the absolute value \fB\fCuids=1-4-10\fR, where 4 is
+calculated by adding 3 (container ID in the \fB\fCuids=\fR mapping)
++ 1 (\fB\fChostID - containerID\fR for the user namespace mapping where
+\fB\fCcontainerID = 1\fR is found).
+
+.PP
+The current implementation doesn't take into account multiple
+user namespace ranges, so it is the caller responsibility to split a
+mapping that overlaps multiple ranges in the user namespace.
```

Comment:
Even though it already states that the current implementation does not support such a case a note that this can (currently) fail silently seems appropriate.

```
sudo mkdir mnt && sudo chown 13999:13999 mnt
sudo podman run --mount "type=bind,src=$(pwd)/mnt,dst=/mnt,idmap=uids=@0-0-14000;gids=@0-0-14000" --userns=auto:size=14000  --rm -it --user 13999:13999 fedora stat -c '%u:%g' /mnt
13999:13999

sudo podman  run --mount "type=bind,src=$(pwd)/mnt,dst=/mnt,idmap=uids=@0-0-14000;gids=@0-0-14000" --uidmap 0:165536:7000 --gidmap 0:165536:7000 --uidmap 7000:200000:7000  --gidmap 7000:200000:7000 --rm -it --user 13999:13999 fedora stat -c '%u:%g' /mnt
65534:65534
```

Reply:
added